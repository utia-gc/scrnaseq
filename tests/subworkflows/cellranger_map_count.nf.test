nextflow_workflow {

    name "Test Workflow Cellranger_Map_Count"
    script "subworkflows/cellranger_map_count.nf"
    workflow "Cellranger_Map_Count"

    test("Cellranger_Map_Count run the `cellranger mkref` and `cellranger count` pipelines from raw inputs -- RefSeq.") {
        // instantiate instances of helper classes
        def reads = new ReadsPE1()
        def refs  = new RefsRefSeq()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = [
                    [
                        sampleName: "${reads.metadata.sampleName}",
                        readType:   "${reads.metadata.readType}"
                    ],
                    [
                        file("${reads.reads[0]}"),
                        file("${reads.reads[1]}")
                    ]
                ]
                input[1] = file("${refs.sequences.genome.decomp}")
                input[2] = file("${refs.annotations.gtf.decomp}")
                """
            }
        }

        then {
            // test workflow status
            assert workflow.success

            // test file for multiqc output is produced
            with(workflow.out.map_count_log) {
                assert size() == 1
                assert get(0) ==~ /^.*\/web_summary.html/
            }
        }

    }

}
