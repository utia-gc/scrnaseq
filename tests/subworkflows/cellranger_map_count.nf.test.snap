{
    "Cellranger_Map_Count run the `cellranger mkref` and `cellranger count` pipelines from raw inputs -- RefSeq.": {
        "content": [
            {
                "0": [
                    "web_summary.html:md5,7d9e6be906c1ea3a0403b8a281fdb19e"
                ],
                "map_count_log": [
                    "web_summary.html:md5,7d9e6be906c1ea3a0403b8a281fdb19e"
                ]
            }
        ],
        "timestamp": "2023-08-24T13:29:40+0000"
    }
}